Source/Core/WebService.swift

```diff
@@ -179,23 +215,36 @@ extension WebService: Session {
 extension WebService {
     /**
      Return an absolute URL string relative to the baseURLString value.
+
+     - parameter string: URL string.
+     - returns: An absolute URL string relative to the value of `baseURLString`.
+     */
+    dynamic public func absoluteURL(_ string: String) -> URL {
+        return constructURL(string, relativeToURL: baseURL)!
+    }
+
+    /**
+     Return an absolute URL string relative to the baseURLString value.
     
      - parameter string: URL string.
-     - returns: An absoulte URL string relative to the value of `baseURLString`.
+     - returns: An absolute URL string relative to the value of `baseURLString`.
     */
-    public func absoluteURLString(_ string: String) -> String {
-        return constructURLString(string, relativeToURLString: baseURLString)
+    dynamic public func absoluteURLString(_ string: String) -> String {
+        return absoluteURL(string).absoluteString
     }
-    
+
     /**
      Return an absolute URL string relative to the baseURLString value.
-    
+
      - parameter string: URL string value.
      - parameter relativeURLString: Value of relative URL string.
      - returns: An absolute URL string.
-    */
-    func constructURLString(_ string: String, relativeToURLString relativeURLString: String) -> String {
-        let relativeURL = URL(string: relativeURLString)
-        return URL(string: string, relativeTo: relativeURL)!.absoluteString
+     */
+    func constructURL(_ string: String, relativeToURL: URL?) -> URL? {
+        guard string != "" else { // if string is empty then just return the baseURL
+            return baseURL
```

Comment:
I think we should just return the `relativeToURL` value here instead of the `baseURL` since that is what is already being passed into function. What do you think?